## 1. State the Hypotheses

 H0: The type of smart home device (Smart Thermostat vs. Smart Light) is independent of customer satisfaction level.
 
 Ha: The type of smart home device and customer satisfaction level are associated

## 2. Compute the Chi-Square Statistic:

 Here Satisfaction of Smart Thermostat is represnted by set A and Smart Light by set B

In [14]:
A=[50,80,60,30,20]
B=[70,100,90,50,50]
## to calculate the mean of the data
mean_a = np.mean(A)
mean_b = np.mean(B)
print(mean_a,mean_b)

48.0 72.0


In [1]:
import numpy as np
import scipy.stats as stats

In [ ]:
# Observed frequencies
observed = np.array([
    [50, 70],
    [80, 100],
    [60, 90],
    [30, 50],
    [20, 50]
])

In [ ]:
# Row and column totals
row_totals = observed.sum(axis=1)
col_totals = observed.sum(axis=0)
grand_total = observed.sum()

In [ ]:
# Expected frequencies
expected = np.outer(row_totals, col_totals) / grand_total

In [ ]:
# Compute Chi-Square Statistic
chi_square_stat = ((observed - expected) ** 2 / expected).sum()

In [ ]:
# Degrees of freedom
rows, cols = observed.shape
df = (rows - 1) * (cols - 1)

In [ ]:
# Critical value at alpha = 0.05
alpha = 0.05
critical_value = stats.chi2.ppf(1 - alpha, df)

In [ ]:
# p-value
p_value = stats.chi2.sf(chi_square_stat, df)

In [ ]:
# Decision
if chi_square_stat > critical_value:
    decision = "Reject the null hypothesis (H0)."
else:
    decision = "Fail to reject the null hypothesis (H0)."

In [6]:
# Output results
print("Chi-Square Statistic:", round(chi_square_stat, 2))
print("Degrees of Freedom:", df)
print("Critical Value:", round(critical_value, 2))
print("p-value:", round(p_value, 4))
print("Decision:", decision)

Chi-Square Statistic: 5.64
Degrees of Freedom: 4
Critical Value: 9.49
p-value: 0.2278
Decision: Fail to reject the null hypothesis (H0).


## 3.The Critical Value Using the alpha of 0.05 and the degrees of freedom is 9.49

## DECISION : 

FAIL TO REJECT THE NULL HYPOTHESIS

## Steps to Perform the  Chi-Square Statistic value

STEP_1: Calculate the expected frequencies for each cell using the formula:
(Row_Total * Column_Total) / (Grand_Total)

STEP_2: computing the chi-Square Statistic

Step_3: Determine the critical Value:
        The degrees of Freedom(df) are calculates as: 
            df = (number_of_rows-1) * (number_of_columns-1)
            Alpha = 0.05

STEP_4: compare the computed X Square value to the critical value
            If X Square Value is Greater than Critical_Value then Reject H0
            otherwise
            Fail to Reject the H0

## Values 

1. Chi-Square Statistic: 5.64
2. Degrees of Freedom: 4
3. Critical Value: 9.49
4. p-value: 0.2278
5. Decision: Fail to reject the null hypothesis (H0).

## CONCLUSION

After performing the Chi-Square test for independence, the results are as follows:
1. Chi-Square Statistic (𝜒2 ): This measures the discrepancy between observed and expected frequencies.
2. Critical Value: The value from the Chi-Square distribution table corresponding to a significance level (𝛼 = 0.05) and degrees of freedom.
3. Decision: Comparing the (𝜒2) statistic to the critical value, we make a conclusion.

Based on the analysis:
1. We fail to reject the null hypothesis (𝐻0).
2. This suggests there is no statistically significant association between the type of smart home device purchased and the customer satisfaction level.
3. Interpretation: Customer satisfaction levels do not significantly differ between Smart Thermostats and Smart Lights.